common/src/main/java/com/microsoft/identity/common/internal/cache/CacheKeyValueDelegate.java

```diff
@@ -144,7 +145,20 @@ public String generateCacheKey(Credential credential) {
         cacheKey = cacheKey.replace(HOME_ACCOUNT_ID, sanitizeNull(credential.getHomeAccountId()));
         cacheKey = cacheKey.replace(ENVIRONMENT, sanitizeNull(credential.getEnvironment()));
         cacheKey = cacheKey.replace(CREDENTIAL_TYPE, sanitizeNull(credential.getCredentialType()));
-        cacheKey = cacheKey.replace(CLIENT_ID, sanitizeNull(credential.getClientId()));
+
+        RefreshToken rt;
+        if ((credential instanceof RefreshToken)
+                && !StringExtensions.isNullOrBlank((rt = (RefreshToken) credential).getFamilyId())) {
+            String familyIdForCacheKey = rt.getFamilyId();
+
+            if (!familyIdForCacheKey.startsWith(FOCI_PREFIX)) {
```

Comment:
I believe this is true for ADAL android... .is this true for unified cache per the spec?

Reply:
This is my understanding...

@oldalton @GTValentine -- Can you confirm? The `foci-<id>` value is supposed to appear in the cache key, correct? Does the `client_id` field of the RT get replaced with this value too? If yes, why do we track both a `client_id` _and_ a `family_id` on RTs if they are the same value?

Currently, the logic to create cache keys just looks for the presence of an `family_id` and, if it exists, replaces the `client_id` in the cache key with its value. The `RefreshToken` written to the cache (as JSON) will contain the `family_id` as `foci-<id>` with the `client_id` as the GUID from the app registration portal.

Reply:
I haven't impleemnted any FOCI logic yet. The whole thing is still in design stage.

Reply:
@iambmelt Based on our previous discussion, we don't replace client_id field with foci-<id> in the schema. In previous ADAL Objective-C cache we did indeed replace client_id with foci-<id> and will have to maintain backward compatibility with it, but it's irrelevant for the perspective of the new schema (but calling out, because legacy ADAL Android cache might have the same logic). 

The family_id field in the schema doesn't need "foci" prefix. 

For the cache keys, yes, there's either client_id or family_id. We're not appending foci prefix in the new cache schema, only in the old one. Please let me know if you feel there's a need for the prefix and I can update our code.

Reply:
If the prefix is not needed I will need to update this pull request to remove it. I don't have a strong preference about whether or not the prefix is present. The only thing I can think of is that it may be more easily human readable if it is there -- if that alone is not a compelling enough reason to prefix it then I say let's just ship it with the digit value